de-lib/src/main/java/org/iplantc/de/client/services/impl/DiskResourceServiceFacadeImpl.java

```diff
@@ -1041,6 +1042,26 @@ public String downloadTemplate(String templateid) {
         return address;
     }
 
+    @Override
+    public void requestHTPathlistFile(HTPathListRequest request, DECallback<String> callback) {
+        StringBuilder address =
+                new StringBuilder(deProperties.getDataMgmtBaseUrl() + "path-list-creator?");
+        address.append("dest=" + URL.encodeQueryString(request.getDest()));
+        address.append("&name-pattern=" + URL.encodeQueryString(request.getPattern()));
+        address.append("&folders-only=" + request.isFoldersOnly());
+        address.append("&recursive=" + true);
+        if (request.getInfoTypes() != null && request.getInfoTypes().size() > 0) {
+            for (String infotype : request.getInfoTypes()) {
+                address.append("&info-type=" + infotype);
+            }
+        }
+        Splittable s = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(request));
+        Splittable spath = StringQuoter.createSplittable();
+        ServiceCallWrapper wrapper =
+                new ServiceCallWrapper(POST, address.toString(), spath.getPayload());
```

Comment:
I'm not seeing how the paths from `Splittable s` are getting into `spath` here (maybe I'm just in a post-lunch haze)?

Maybe it would be better to use a `HasPaths` AutoBean here to generate the payload?
Then you could call `HasPaths#setPaths` with `request.getPaths`.
Or maybe even encode `request` directly into a `HasPaths` AutoBean?